### ISSUE TITLE
[FEATURE] enable output_margin, pred_contribs in sklearn API

### ISSUE THREAD

**Issue by pseudotensor** [dmlc/xgboost#6977]
We maintain a slightly adjusted version of xgboost for our own purposes, but one thing that is common are some trivial features that are not exposed.  I've mentioned this to @trivialfis before, but wanted to bring up since it's been several months and we are applying latest changes to ours.

Here's what I'd recommend:

1) sklearn.py pred_leaf, pred_contribs, approximate_contribs

diffs here:
(some here, but not all): https://github.com/h2oai/xgboost/commit/bfff9ab50084eaf3311379d7ae194cca3b6f94f0
i.e. at: https://github.com/h2oai/xgboost/blob/bfff9ab50084eaf3311379d7ae194cca3b6f94f0/python-package/xgboost/sklearn.py
i.e. API related ones (not label encoder ones): ![image](https://user-images.githubusercontent.com/2249614/119048238-cf95a300-b973-11eb-8dc2-593a4e731fad.png)

2) output_margin de-hard-code and use:

https://github.com/h2oai/xgboost/commit/355e415c1c8a5a148b5a4fef545d0ccc7b44b29c

As-is, output_margin now can no longer be chosen independently.  And pred_contribs etc. are not accessible still.

**Comment by hcho3** [comment]
@pseudotensor Would you like to submit a pull request? Let's get the change into the upstream so that it'd be easier for you to sync your fork with the upstream.

**Comment by trivialfis** [comment]
Predict leaf is in the `apply` method.
